python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -290,146 +292,157 @@ def _makeBoxRegion(BBox, wcs, BBoxPadding):
         outerLocalBBox = geom.Box2D(BBox)
         innerLocalBBox = geom.Box2D(BBox)
 
-        # Grow the bounding box to make sure the spherical geometry bbox will contain
-        # the same region, as non-padded boxes may contain different regions because of optical distortion.
-        # Also create an inner region that is sure to be inside the bbox
+        # Grow the bounding box to allow for effects not fully captured by the
+        # wcs provided (which represents the current best-guess wcs solution
+        # associated with the dataset for which the calibration is to be
+        # computed using the loaded and trimmed reference catalog being defined
+        # here.  These effects could include pointing errors and/or an
+        # insufficient optical distorition model for the instrument.
+        # Also create an inner region that is sure to be inside the bbox.
         outerLocalBBox.grow(BBoxPadding)
         innerLocalBBox.grow(-1*BBoxPadding)
 
-        # Handle the fact that the inner bounding box shrunk to a zero sized region in at least one
-        # dimension, in which case all reference catalogs must be checked fully against the input
-        # bounding box
+        # Handle the case where the inner bounding box shrank to a zero sized
+        # region (which will be the case if the shrunken size of either
+        # dimension is less than or equal to zero).  In this case, the inner
+        # bounding box is set to the original input bounding box.  This is
+        # probably not the best way to handle an empty inner bounding box, but
+        # it is what the calling code currently expects.
         if innerLocalBBox.getDimensions() == geom.Extent2D(0, 0):
-            innerSkyRegion = sphgeom.Box()
-        else:
-            innerBoxCorners = innerLocalBBox.getCorners()
-            innerSphCorners = [wcs.pixelToSky(corner).getVector() for corner in innerBoxCorners]
+            innerLocalBBox = geom.Box2D(BBox)
 
-            innerSkyRegion = sphgeom.ConvexPolygon(innerSphCorners)
+        # Convert the corners of the bounding boxes to sky coordinates.
+        innerBoxCorners = innerLocalBBox.getCorners()
+        innerSphCorners = [wcs.pixelToSky(corner).getVector() for corner in innerBoxCorners]
+        innerSkyRegion = sphgeom.ConvexPolygon(innerSphCorners)
 
-        # Convert the corners of the box to sky coordinates
         outerBoxCorners = outerLocalBBox.getCorners()
         outerSphCorners = [wcs.pixelToSky(corner).getVector() for corner in outerBoxCorners]
-
         outerSkyRegion = sphgeom.ConvexPolygon(outerSphCorners)
 
         return innerSkyRegion, outerSkyRegion, innerSphCorners, outerSphCorners
 
-    def loadPixelBox(self, bbox, wcs, filterName=None, epoch=None, photoCalib=None, bboxPadding=100):
-        """Load reference objects that are within a pixel-based rectangular region
+    def loadPixelBox(self, bbox, wcs, filterName=None, epoch=None, photoCalib=None):
```

Comment:
This needs to change back, see below (feel free to ask more questions if what I wrote was not clear)

Reply:
See comments below.

Reply:
Effectively reverted.